Yosemite/Yosemite/Stores/AccountStore.swift

```diff
@@ -195,6 +195,13 @@ extension AccountStore {
     func upsertStoredSitesInBackground(readOnlySites: [Networking.Site], onCompletion: @escaping () -> Void) {
         let derivedStorage = sharedDerivedStorage
         derivedStorage.perform {
+            // Deletes sites in storage that are not in `readOnlySites`.
+            let storageSites = derivedStorage.loadAllSites()
+            storageSites?.filter { readOnlySites.map { $0.siteID }.contains($0.siteID) == false }
+                .forEach { remotelyDeletedSite in
+                    derivedStorage.deleteObject(remotelyDeletedSite)
+                }
```

Comment:
Do you think moving the map outside would be better for readability? I'm not sure. 🤔 

```suggestion
            let readOnlySiteIDs = readOnlySites.map(\.siteID)
            storageSites.filter { readOnlySiteIDs.contains($0.siteID) == false }
                .forEach { remotelyDeletedSite in
                    derivedStorage.deleteObject(remotelyDeletedSite)
                }
```

Reply:
yup! updated in https://github.com/woocommerce/woocommerce-ios/pull/5241/commits/2e841d6ad663298c0f31d41cdfd7e866225b855b